translate/src/modules/user/components/UserMenu.tsx

```diff
@@ -37,6 +40,57 @@ export function UserMenuDialog({
   const ref = useRef<HTMLUListElement>(null);
   useOnDiscard(ref, onDiscard);
 
+  const [theme, setTheme] = useState<string>(
+    () => document.body.getAttribute('data-theme') || 'dark',
+  );
+
+  function getSystemTheme() {
+    if (
+      window.matchMedia &&
+      window.matchMedia('(prefers-color-scheme: dark)').matches
+    ) {
+      return 'dark';
+    } else {
+      return 'light';
+    }
+  }
+
+  useEffect(() => {
+    function applyTheme(newTheme: string) {
+      if (newTheme === 'system') {
+        newTheme = getSystemTheme();
+      }
+      document.body.classList.remove(
+        'dark-theme',
+        'light-theme',
+        'system-theme',
+      );
+      document.body.classList.add(`${newTheme}-theme`);
+    }
+
+    const mediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
+    function handleThemeChange(e: MediaQueryListEvent) {
+      let userThemeSetting = document.body.getAttribute('data-theme') || 'dark';
+
+      if (userThemeSetting === 'system') {
+        applyTheme(e.matches ? 'dark' : 'light');
+      }
+    }
+
+    mediaQuery.addEventListener('change', handleThemeChange);
+
+    applyTheme(theme);
+
+    return () => {
+      mediaQuery.removeEventListener('change', handleThemeChange);
+    };
+  }, [theme]);
+
+  const handleThemeButtonClick = (selectedTheme: string) => {
+    setTheme(selectedTheme); // Update the local state
```

Comment:
State already gets updated through the `onThemeChange()` method, which ends up calling the `saveTheme()` action. 

The only thing we should call here is applyTheme(), ideally from the same place that [Theme.tsx](https://github.com/mozilla/pontoon/blob/91977b087f6fc36a5d821eec1e37091471a4bf53/translate/src/context/Theme.tsx) should be calling it. Then you can remove lines 43-87.

Reply:
We should call `applyTheme()` here.